addons/binding/org.openhab.binding.amazondashbutton/src/main/java/org/openhab/binding/amazondashbutton/internal/config/AmazonDashButtonConfigDescriptionProvider.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.amazondashbutton.internal.config;
+
+import static org.openhab.binding.amazondashbutton.AmazonDashButtonBindingConstants.DASH_BUTTON_THING_TYPE;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.core.ConfigDescription;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter.Type;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameterBuilder;
+import org.eclipse.smarthome.config.core.ConfigDescriptionProvider;
+import org.eclipse.smarthome.config.core.ParameterOption;
+import org.eclipse.smarthome.config.core.i18n.ConfigI18nLocalizationService;
+import org.eclipse.smarthome.core.common.osgi.ServiceBinder.Bind;
+import org.eclipse.smarthome.core.common.osgi.ServiceBinder.Unbind;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.openhab.binding.amazondashbutton.internal.PcapUtil;
+import org.osgi.framework.FrameworkUtil;
+import org.pcap4j.core.PcapAddress;
+import org.pcap4j.core.PcapNetworkInterface;
+
+/**
+ * The {@link AmazonDashButtonConfigDescriptionProvider} provides {@link ConfigDescription}s which can not be modeled by
+ * using the thing-description XML schema.
+ *
+ * In particular it determines the available pcap network interfaces in order to provide an option list.
+ *
+ * @author Oliver Libutzki - Initial contribution
+ *
+ */
+public class AmazonDashButtonConfigDescriptionProvider implements ConfigDescriptionProvider {
```

Comment:
I don't think you actually need to construct the config descriptions - you can do them in XML.
All you want is to implement a ConfigOptionProvider in order to offer some selection options.